src/javascript/request-composer/components/request-composer.js

```diff
@@ -0,0 +1,614 @@
+   */
+  handleParamChange = ({target: {name, value}}) => {
+    this.props.actions.updateParams({[name]: value});
+
+    if (name == 'metrics' || name == 'dimensions') {
+      this.props.actions.updateSortOptions();
+    }
+  }
+
+
+  /**
+   * Invoked when a user changes the request type.
+   * @param Int index The index of the selcected request tab.
+   * @param Int last The index of the last selected request tab.
+   */
+  handleRequestChange = (index) => {
+    this.props.actions.updateSettings({requestType: REQUEST_TYPES[index]});
+  }
+
+
+  /**
+   * Invoked when a user clicks on the segment definition checkbox.
+   * @param {{target: Element}} The React event.
+   */
+  handleSegmentDefinitionToggle = ({target: {checked: useDefinition}}) => {
+    let {actions} = this.props;
+    let {segment} = this.props.params;
+
+    actions.updateSettings({useDefinition});
+    actions.updateSegmentsOptions(useDefinition);
+
+    if (segment) {
+      actions.swapSegmentIdAndDefinition(segment, useDefinition);
+    }
+  }
+
+
+  /**
+   * Invoked when a user submits the <QueryForm>.
+   * @param {Event|Object} e The native or React event.
+   */
+  handleSubmit = async (e) => {
+    e.preventDefault();
+    let {actions, params, settings} = this.props;
+    let paramsClone = {...params};
+
+    actions.setQueryState(true);
+
+    let trackableParamData = Object.keys(paramsClone).map((key) =>
+        PARAMS_TO_TRACK.includes(key) ? `${key}=${paramsClone[key]}` : key)
+        .join('&');
+
+    // Set it on the tracker so it gets sent with all Query Explorer hits.
+    gaAll('set', 'dimension2', trackableParamData);
+
+    let request = composeRequest(params, settings);
+    gapi.client.analyticsreporting.reports.batchGet(request
+      ).then(function(response) {
+        actions.updateResponse(response);
+        actions.updateSettings({
+          'responseType': settings.requestType,
+          'responseCohortSize': params.cohortSize
+      });
+      }, function(response) {
+        AlertDispatcher.addOnce({
+          title: 'Oops, there was an error',
+          message: response.result.error.message
+        });
+        actions.updateResponse(response);
+    });
+
+    actions.setQueryState(false);
+  }
+
+
+  /**
+   * React lifecycyle method below:
+   * http://facebook.github.io/react/docs/component-specs.html
+   * ---------------------------------------------------------
+   */
+  render() {
+
+    let {
+      isQuerying,
+      params,
+      response,
+      settings,
+      select2Options
+    } = this.props;
+
+    let formControlClass = 'FormControl FormControl--inline';
+    let formActionClass = formControlClass + ' FormControl--action';
+    let requiredFormControlClass = formControlClass +' FormControl--required';
+    let maximumSelectionSize = 2;
+
+    return (
+      <div>
+        <h3 className="H3--underline">Select a view</h3>
+
+        <ViewSelector
+          viewId={params.viewId}
```

Comment:
This isn't actually doing anything since the `ViewSelector` component isn't currently configured to initialize with a `viewId` prop. You could add it though by updating this logic: https://github.com/mcohoon/ga-dev-tools/blob/request-composer/src/javascript/components/view-selector.js#L62-L64